src/test/java/codeu/model/store/basic/EventStoreTest.java

```diff
@@ -0,0 +1,89 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.Event;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class EventStoreTest {
+  
+  private EventStore eventStore;
+  private PersistentStorageAgent mockPersistentStorageAgent;
+
+  private final List<String> INFORMATION_ONE = 
+      new ArrayList<>(Arrays.asList("user_one"));
+  private final List<String> INFORMATION_TWO = 
+      new ArrayList<>(Arrays.asList("user_two", "about_me"));
+  private final List<String> INFORMATION_THREE = 
+      new ArrayList<>(Arrays.asList("user_three", "conversation_title"));
+  private final List<String> INFORMATION_FOUR = 
+      new ArrayList<>(Arrays.asList("user_four", "conversation_title", "message_content"));
+
+  private final Event EVENT_ONE = 
+      new Event(UUID.randomUUID(), "User", Instant.now(), INFORMATION_ONE);
+  private final Event EVENT_TWO = 
+      new Event(UUID.randomUUID(), "About Me", Instant.now(), INFORMATION_TWO);
+  private final Event EVENT_THREE = 
+      new Event(UUID.randomUUID(), "Conversation", Instant.now(), INFORMATION_THREE);
+  private final Event EVENT_FOUR = 
+      new Event(UUID.randomUUID(), "Message", Instant.now(), INFORMATION_FOUR);
+
+  @Before
+  public void setUp() {
+    mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+    eventStore = EventStore.getTestInstance(mockPersistentStorageAgent);
+
+    final List<Event> eventList = new ArrayList<>();
+    eventList.add(EVENT_ONE);
+    eventList.add(EVENT_TWO);
+    eventList.add(EVENT_THREE);
+    eventList.add(EVENT_FOUR);
+    eventStore.setEvents(eventList);
+  }
+
+  @Test
+  public void testGetEvent_byId_found() {
+    Event resultEvent = eventStore.getEvent(EVENT_ONE.getId());
+
+    assertEquals(EVENT_ONE, resultEvent);
+  }
+
+  @Test
+  public void testGetEvent_byId_notFound() {
+    Event resultEvent = eventStore.getEvent(UUID.randomUUID());
+
+    Assert.assertNull(resultEvent);
+  }
+
+  @Test
+  public void testAddEvent() {
+    UUID eventId = UUID.randomUUID();
+    List<String> testInformation = 
+        new ArrayList<>(
+          Arrays.asList(
+            "test_username", 
+            "test_conversation_title", 
+            "test_message_content"));
+    Event inputEvent = new Event(eventId, "Message", Instant.now(), testInformation);
+
+    eventStore.addEvent(inputEvent);
+    Event resultEvent = eventStore.getEvent(eventId);
+
+    assertEquals(inputEvent, resultEvent);
+    Mockito.verify(mockPersistentStorageAgent).writeThrough(inputEvent);
+  }
+
+  private void assertEquals(Event expectedEvent, Event actualEvent) {
```

Comment:
Why not have an equals method in the event class if you're going to make one here anyway. Maybe it could be helpful somehow in the future. and this way you can also use it when comparing those methods in PersistentDataStoreTest.java

Reply:
I added it here to be consistent with the other test files that were in the original code base? Do you guys think I should still move this function to the Event class?

Reply:
Well if it were, you could use it in the PersistentDataStoreTest.java as well to compare the events instead of a for loop comparing each property of the event. But it's not that big of deal, maybe it is better to keep it consistent with the others.